pkg/render/intrusiondetection/dpi/dpi_test.go

```diff
@@ -243,6 +268,47 @@ var _ = Describe("DPI rendering tests", func() {
 		validateDPIComponents(resources, false)
 	})
 
+	It("should render all resources for deep packet inspection with default resource requirements for a management cluster without DPI resources", func() {
+		cfg = &dpi.DPIConfig{
+			IntrusionDetection: ids,
+			Installation:       installation,
+			TyphaNodeTLS:       typhaNodeTLS,
+			PullSecrets:        pullSecrets,
+			Openshift:          false,
+			HasNoLicense:       false,
+			HasNoDPIResource:   true,
+			ManagementCluster:  true,
+			ESClusterConfig:    esConfigMap,
+			ClusterDomain:      dns.DefaultClusterDomain,
+			DPICertSecret:      dpiCertSecret,
+		}
+		component := dpi.DPI(cfg)
+
+		resources, _ := component.Objects()
```

Comment:
Do we have a test that confirms the `objectsToDelete` is populated correctly for managed clusters and standalone clusters without DPI?

Reply:
For standalone, we already have tests for it. 
https://github.com/tigera/operator/blob/master/pkg/render/intrusiondetection/dpi/dpi_test.go#L294
https://github.com/tigera/operator/blob/master/pkg/render/intrusiondetection/dpi/dpi_test.go#L319

I can add one for managed also.

Reply:
I think for a managed cluster we never create the cluster role and cluster rolebinding, but we include them for deletion.